src/endpoints/transactions/transaction.service.ts

```diff
@@ -128,6 +128,27 @@ export class TransactionService {
 
     let transactions: Transaction[] = [];
 
+    if (filter.hashes) {
+      const txHashes: string[] = filter.hashes.split(',');
+      const elasticHashes = elasticTransactions.map(({txHash}) => txHash);
+      const missingHashes: string[] = txHashes.findMissingElements(elasticHashes);
+      let gatewayTransactions = await Promise.all(
+        missingHashes
+          .map(async (txHash) => {
+            const gatewayTransaction = await this.tryGetTransactionFromGateway(txHash);
```

Comment:
I would add here a parameter so that inside the tryGetTransactionFromGateway the tryGetTransactionFromElasticBySenderAndNonce function is not called